site/app/libraries/database/DatabaseQueries.php

```diff
@@ -7785,68 +7785,61 @@ public function openPoll($poll_id) {
         $this->course_db->query("UPDATE polls SET status = 'open' where poll_id = ?", [$poll_id]);
     }
 
-    public function getPolls() {
+    public function getPolls(): array {
         $polls = [];
         $this->course_db->query("SELECT * from polls order by poll_id ASC");
         $polls_rows = $this->course_db->rows();
-        $user = $this->core->getUser()->getId();
 
         foreach ($polls_rows as $row) {
-            $polls[] = $this->getPoll($row["poll_id"]);
+            $polls[] = new PollModel($this->core, $row["poll_id"], $row["name"], $row["question"], $row["question_type"], $row["status"], $row["release_date"], $row["image_path"], $row["release_histogram"]);
         }
 
         return $polls;
```

Comment:
Let's add some optional joins here with the responses, answers, and user_responses so that the full PollModel can be populated initially if we know we're going to need it later. This could either probably be in this function with a boolean or a separate function.